plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/local/LocalFileProvider.java

```diff
@@ -114,6 +115,28 @@ public class LocalFileProvider extends BaseFileProvider<LocalFile> {
     return getFiles( file, filters );
   }
 
+  @Override
+  public List<LocalFile> searchFiles( LocalFile file, String filters, String searchString, VariableSpace space )
+    throws FileException {
+    List<LocalFile> files = new ArrayList<>();
+    try ( Stream<Path> pathStream = Files.walk( Paths.get( file.getPath() ) ) ) {
+      pathStream.forEach( path -> {
+        String name = path.getFileName().toString();
```

Comment:
This line throws a null pointer exception on Linux when trying to search from the root directory /

Reply:
Solved the issue and pushed the changes